validation/citrus-validation-json/src/main/java/org/citrusframework/validation/json/JsonElementValidator.java

```diff
@@ -110,7 +110,10 @@ private void validateJSONArray(JsonElementValidator validator, JsonElementValida
             validateSameSize(control.getJsonPath(), arrayControl.expected, arrayControl.actual);
         }
         for (int i = 0; i < arrayControl.expected.size(); i++) {
-            if (!isAnyValidItemInActualArray(validator, arrayControl, i)) {
+            var isValid = strict
+                    ? isValidItem(new JsonElementValidatorItem<>(i, arrayControl.actual.get(i), arrayControl.expected.get(i)).parent(control), validator)
+                    : isAnyValidItemInActualArray(validator, arrayControl, i);
```

Comment:
In general I'd say `isAnyValidItemInActualArray` is not sufficient to determine correct order of an array in non strict mode.  

This should be revrealed by:

```
control = [1, 1 ,2, 1]
actual= [1, 2, 2, 2]
```

To be non breaking, you should implement the behaviour as it was before. Strict was only used to verify the number of elements in the array. So if the received array had trailing elements, this was OK in non strict mode. But the order was still verified for all elements of the control array. If we change this behaviour as proposed, we are risking false positives. Also, in general, this would not support our use case. We need to be able to individually decide whether we want to neglect sorting. Strict/non strict works on a global level.

Reply:
> It's essential to remember that strict mode should not dictate how the order of an array is validated.

don't you think this strange? 👯 that makes no sense. strict is strict. period.

Reply:
I don't know what you are refering to to. Strict was in the past only used to tell the validator whether validation should fail on additional elements. This holds true for objects and also for arrays. 

To makt this clear: up to my knowledge there has never been a way to tell the validator to ignore ordering of arrays.

Reply:
Array Order now always gets validated (as before #1102).

And mentioned Testcase was added